site/gdocs/ArticleBlock.tsx

```diff
@@ -272,45 +275,55 @@ export default function ArticleBlock({
                                     getLayout("heading", containerType),
                                     {
-                                        "has-supertitle": supertitleText,
+                                        "has-supertitle": supertitle
+                                            ? spansToUnformattedPlainText(
+                                                  supertitle
+                                              )
+                                            : "",
                                     }
                                 )}
                                 id={id}
                             >
-                                {supertitleText ? (
+                                {supertitle ? (
                                     <div className="article-block__heading-supertitle overline-black-caps">
-                                        {supertitleText}
+                                        {renderSpans(supertitle)}
                                     </div>
                                 ) : null}
-                                {text}
+                                {renderSpans(text)}
                             </h2>
                         </>
                     )
                 })
                 .with({ type: "heading", level: 3 }, (block) => {
-                    const {
-                        supertitle,
-                        text: { text },
-                    } = block
-                    const supertitleText = supertitle?.text || ""
-                    const id = urlSlug(`${supertitleText}-${text}`)
+                    const { supertitle, text } = block
+                    const id = supertitle
+                        ? urlSlug(
+                              `${spansToUnformattedPlainText(
+                                  supertitle
+                              )}-${spansToUnformattedPlainText(text)}`
+                          )
+                        : urlSlug(spansToUnformattedPlainText(block.text))
                     return (
                         <h3
                             className={cx(
                                 "h2-bold",
                                 getLayout("heading", containerType),
                                 {
-                                    "has-supertitle": supertitleText,
+                                    "has-supertitle": supertitle
+                                        ? spansToUnformattedPlainText(
+                                              supertitle
+                                          )
+                                        : undefined,
                                 }
                             )}
                             id={id}
                         >
-                            {supertitleText ? (
+                            {supertitle ? (
                                 <div className="article-block__heading-supertitle overline-black-caps">
```

Comment:
You're right; h3's are only allowed [phrasing content](https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#phrasing_content). To make the HTML structure valid we should switch this to a span.